From a38df540af6b79a319e1f3daf9f0aeb9b93940b7 Mon Sep 17 00:00:00 2001
From: Colin Xu <colin.xu@intel.com>
Date: Tue, 25 Aug 2020 16:15:33 +0800
Subject: [PATCH 23/24] drm/i915/gvt: Fix incorrect skipping
 enable_rpm_wakeref_asserts.

gvt suspend/resume routine shouldn't skip i915 pm logic.
Invoke gvt suspend/resume routine without early return.

Signed-off-by: Colin Xu <colin.xu@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.c | 21 +++++++--------------
 1 file changed, 7 insertions(+), 14 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index 10d31c803dab..a1e1e41a8fdf 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -1798,8 +1798,9 @@ static int i915_drm_suspend(struct drm_device *dev)
 	drm_kms_helper_poll_disable(dev);
 
 #if IS_ENABLED(CONFIG_DRM_I915_GVT)
-	if (dev_priv->gvt)
+	if (dev_priv->gvt) {
 		intel_gvt_pm_suspend(dev_priv->gvt);
+	}
 #endif
 
 	pci_save_state(pdev);
@@ -1990,7 +1991,7 @@ static int i915_drm_resume(struct drm_device *dev)
 
 #if IS_ENABLED(CONFIG_DRM_I915_GVT)
 	if (dev_priv->gvt) {
-		return intel_gvt_pm_resume(dev_priv->gvt);
+		intel_gvt_pm_resume(dev_priv->gvt);
 	}
 #endif
 
@@ -2072,12 +2073,8 @@ static int i915_drm_resume_early(struct drm_device *dev)
 	intel_gt_sanitize(&dev_priv->gt, true);
 
 #if IS_ENABLED(CONFIG_DRM_I915_GVT)
-	if (!ret) {
-		if (dev_priv->gvt) {
-			ret = intel_gvt_pm_early_resume(dev_priv->gvt);
-			if (ret)
-				return ret;
-		}
+	if (dev_priv->gvt) {
+		intel_gvt_pm_early_resume(dev_priv->gvt);
 	}
 #endif
 
@@ -2747,12 +2744,8 @@ static int intel_runtime_resume(struct device *kdev)
 	intel_enable_ipc(dev_priv);
 
 #if IS_ENABLED(CONFIG_DRM_I915_GVT)
-	if (!ret) {
-		if (dev_priv->gvt) {
-			ret = intel_gvt_pm_early_resume(dev_priv->gvt);
-			if (ret)
-				return ret;
-		}
+	if (dev_priv->gvt) {
+		intel_gvt_pm_early_resume(dev_priv->gvt);
 	}
 #endif
 
-- 
2.17.1

